File: stonex_ops/tools/tenant_list.py
"""tenant_list: list all tenants with identity, status, and connection-state summary."""

from __future__ import annotations

import json
from typing import Any

from stonex_ops.executor import execute
from stonex_ops.whitelist import TenantList


async def run(stonx_bin: str, env: str, path: str) -> dict[str, Any]:
    output = await execute(stonx_bin, env, path, TenantList())
    parsed = json.loads(output)
    return {"tenants": parsed.get("tenants", [])}

File: src/tools/tenant_list.rs
//! tenant_list:列出所有租户的概览信息。
//!
//! 返回每个租户的 identity、ops_tag、状态和连接状态。

use serde::Serialize;
use std::path::PathBuf;

use crate::executor;
use crate::whitelist::AllowedCommand;

#[derive(Debug, Serialize)]
pub(crate) struct TenantListResult {
    pub tenants: Vec<serde_json::Value>,
}

pub(crate) async fn run(
    stonx_bin: PathBuf,
    env: &str,
    path: &str,
) -> Result<TenantListResult, String> {
    let cmd = AllowedCommand::TenantList;
    let output = executor::execute(stonx_bin, env, path, &cmd)
        .await
        .map_err(|e| format!("tenant list failed: {e}"))?;

    let tenants: Vec<serde_json::Value> = serde_json::from_str(&output.stdout)
        .map_err(|e| format!("parse tenant list json: {e}"))?;

    Ok(TenantListResult { tenants })
}
